hello/wat/hello.ts

```diff
@@ -0,0 +1,194 @@
+        {
+            tag: 'ComponentTypeDeclarationImport',
+            value: {
+                tag: 'ComponentImport',
+                name: {
+                    tag: 'ComponentExternNameKebab',
+                    name: "import-type-city-info"
+                },
+                ty: {
+                    value: 
+                    {
+                        tag: 'TypeBoundsEq',
+                        value: 0
+                    }
+                },
+            }
+        },
+        {
+            tag: 'ComponentTypeDeclarationImport',
+            value: {
+                tag: 'ComponentImport',
+                name: {
+                    tag: 'ComponentExternNameKebab',
+                    name: "import-type-city-info0"
+                },
+                ty: {
+                    value: 
+                    {
+                        tag: 'TypeBoundsEq',
+                        value: 1
+                    }
+                },
+            }
+        },
+        {
+            tag: 'ComponentTypeDeclarationType',
+            value: {
+                tag: 'ComponentTypeFunc',
+                value:
+                {
+                    params: [
+                        [
+                            "info",
+                            {
+                                tag: 'ComponentValTypePrimitive',
+                                value: PrimitiveValType.S32 // no guarantee that 2 is int32
+                            }
+                        ]
+                    ],
+                    results:
+                    {
+                        tag: 'ComponentFuncResultUnnamed',
+                        value: undefined, // there is no info about the results
+                    },
+                }
+            },
+        },
+        {
+            tag: 'ComponentTypeDeclarationImport',
+            value: {
+                tag: 'ComponentImport',
+                name: {
+                    tag: 'ComponentExternNameKebab',
+                    name: "import-func-run"
+                },
+                ty: { value: 3 }, // ComponentTypeRefFunc is it connected with "(func (;0;) (type 3))"?
+            }
+        },
+        {
+            name: {
+                tag: 'ComponentExternNameKebab',
+                name: "city-info"
+            },
+            ty: {
+                value: {
+                    tag: 'TypeBoundsEq',
+                    value: 1 // you sure? Here we don't have eq
+                }
+            },
+        },
+        {
+            tag: 'ComponentTypeDeclarationType',
+            value: {
+                tag: 'ComponentTypeFunc',
+                value:
+                {
+                    params: [
+                        [
+                            "info",
+                            {
+                                tag: 'ComponentValTypePrimitive',
+                                value: PrimitiveValType.S32 // no guarantee that 4 is int32
```

Comment:
`s32` is `int32` s stands for signed

Reply:
We have "(func (param "info" 4))" and 4 can be whatever type, I guess it's int32 but there's no guarantee. That's what it means.